src/utilities/compile.cpp

```diff
@@ -54,7 +54,7 @@ std::shared_ptr<Program> CompileFromSource(
 
   // For specific devices add a global synchronisation barrier to the GEMM kernel to optimize
   // performance through better cache behaviour
-  if (device.IsARM() && device.IsGPU()) {
+  if ((device.IsARM() && device.IsGPU()) || device.IsQualcomm()) {
```

Comment:
Just to be sure (you probably know best), but is it sufficient to say `IsQualcomm` and not also make sure that it is a GPU? Not sure what other OpenCL Qualcomm devices there are, but they might not all require/want all these defines?